packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -111,12 +157,104 @@ export class PathNode extends PathShared<FieldKey> {
  * Thus this can be thought of as a sparse copy of the subset of trees which are used as anchors
  * (and thus need parent paths).
  */
-export class PathCollection extends PathShared<RootField> {
-    public constructor() {
-        super();
+class PathNode implements UpPath {
+    /**
+     * Number of references to this from external sources (ex: `Anchors` via `AnchorSet`.)
+     */
+    private refCount = 1;
+
+    /**
+     * Construct a PathNode with refcount 1.
+     * @param anchorSet - used to determine if this PathNode is already part a specific anchorSet
+     * to early out UpPath walking.
+     */
+    public constructor(
+        public readonly anchorSet: AnchorSet,
+        public parentField: FieldKey,
+        public parentIndex: number,
+        public parentPath: PathNode | undefined) {}
+    // PathNode arrays are kept sorted by index for efficient search.
+    protected readonly children: Map<FieldKey, PathNode[]> = new Map();
+    // public constructor() {}
+
+    public get parent(): UpPath | undefined {
+        // Root PathNode corresponds to the undefined root for UpPath.
+        if (this.parentPath?.parentPath === undefined) {
+            return undefined;
+        }
```

Comment:
What is the prupose of this if block? I'm having trouble seeing why `return this.parentPath;` is not sufficient.

Reply:
I have refactored the code and comment to make this more clear (it still has the same behavior as before). Let me know if this answers your question.